jobs/involuntary-dissolutions/involuntary_dissolutions.py

```diff
@@ -239,16 +239,34 @@ def stage_2_process(app: Flask):
     stage_2_delay = timedelta(days=app.config.get('STAGE_2_DELAY'))
 
     for batch_processing in batch_processings:
-        # Check if email or letter furnishing entry has been completed. If not, do not transition to stage 2.
+        # check furnishing entries - 2 scenarios
+        # - 1. if email processed, and mail processed(after 5 business days still not in GS)
+        # - 2.only available to send mail out, and it's processed
+        # If not, do not transition to stage 2.
         furnishings = Furnishing.find_by(
             batch_id=batch_processing.batch_id,
             business_id=batch_processing.business_id
         )
-        furnishing_entry_completed = any(
-            furnishing.furnishing_type in (Furnishing.FurnishingType.EMAIL, Furnishing.FurnishingType.MAIL)
+        email_processed = any(
+            furnishing.furnishing_type == Furnishing.FurnishingType.EMAIL
             and furnishing.status == Furnishing.FurnishingStatus.PROCESSED
             for furnishing in furnishings
         )
+        # if SFTP function is off, we expect the mail status will be QUEUED
+        expected_mail_status = Furnishing.FurnishingStatus.PROCESSED
+        if flags.is_on('disable-dissolution-sftp-bcmail'):
+            expected_mail_status = Furnishing.FurnishingStatus.QUEUED
```

Comment:
We will turn FF on and off for testing purpose, so you should make sure that both `PROCESSED` and `QUEUED` pass the check. For example, the mail furnishing is set processed in stage 1 with disable-FF off, and then we turn this flag on and move it into stage 2 and this is also expected to pass the check.

Reply:
good call! thanks for pointing it out 👍

Reply:
adjusted the logic, now it appends the QUEUED status for checking when FF is on

Reply:
Think you can also update tests a bit 👀

Reply:
haha, you read my mind, I'm adding tests, like now 😄